src/creature.cpp

```diff
@@ -135,27 +135,32 @@ void Creature::onThink(uint32_t interval)
 		blockTicks = 0;
 	}
 
-	if (followCreature) {
-		walkUpdateTicks += interval;
-		if (forceUpdateFollowPath || walkUpdateTicks >= 2000) {
-			walkUpdateTicks = 0;
-			forceUpdateFollowPath = false;
-			isUpdatingPath = true;
+	const Position& position = getPosition();
+	for (auto* creature : followers) {
+		const Position& followerPosition = creature->getPosition();
+		uint16_t distance = position.getDistanceX(followerPosition) + position.getDistanceY(followerPosition);
+		if (distance >= Map::maxViewportX + Map::maxViewportY || getPosition().z != creature->getPosition().z) {
+			removeFollower(creature);
```

Comment:
It seems this particular piece of code's only purpose is to remove followers when out of viewport range. Since doing this in onThink would cause a delay in when the actual removal takes place, as well as add un-necessary load to onThink, I believe a more event driven approach best... this particular removal logic should instead be relocated to the onMove.